### ISSUE TITLE
Bug: Scrolling stops working on small screens after clicking Sheet Overlay

### ISSUE THREAD

**Issue by matthewmturner** [tamagui/tamagui#1224]
### Current Behavior

Im using Dialog and Adapting to Sheet on small screens, but only on small screens when clicking the overlay to close the sheet all scrolling stops working.  I think the functionality behind disableRemoveScroll isnt being used correctly in this case - when i set that to true i get the expected behavior and i can scroll again.

### Expected Behavior

After clicking overlay on small screens to dismiss Sheet I am able to scroll again

### Tamagui Version

```markdown
1.29.1
```

### Reproduction

```markdown
Below is the core code I have now.  I hope to produce a sandbox tomorrow.


export function SessionRosterRegistration({
	groups,
}: SessionRosterRegistrationProps) {
	const [isRegistrationVisible, setIsRegistrationVisible] = useState(false);
	const { useSessionDetails } = useSessionRoster();
	const { listing } = useSessionDetails();
	const { control, handleSubmit } = useForm<RegistrationInfo>();
	const onSubmit = (regisrationInfo: RegistrationInfo) => {
		handleRegistration(regisrationInfo);
		setIsRegistrationVisible(false);
		const listingPassword = readSessionPasswordFromStorage(listing.id.uuid);
		fetchRoster(listingPassword);
	};
	const description =
		"Enter the following information to sign up.\nClick save when done.";

	return (
		<Dialog open={isRegistrationVisible} disableRemoveScroll={true}>
			<Button
				iconAfter={PlusCircle}
				onPress={() => {
					setIsRegistrationVisible(true);
				}}
			>
				Register
			</Button>

			<Adapt when="sm" platform="touch">
				<Sheet zIndex={200000} modal>
					<Sheet.Frame padding="$4" space>
						<Form onSubmit={handleSubmit(onSubmit)}>
							<Adapt.Contents />
						</Form>
					</Sheet.Frame>
					<Sheet.Overlay />
				</Sheet>
			</Adapt>

			<Dialog.Portal>
				<RegistrationBackground />
				<Form onSubmit={handleSubmit(onSubmit)}>
					<RegistrationContent
						description={description}
						groups={groups}
						control={control}
						setIsRegistrationVisible={setIsRegistrationVisible}
					/>
				</Form>
			</Dialog.Portal>
		</Dialog>
	);
}
```
```


### System Info

```markdown
System:
    OS: macOS 13.0
    CPU: (8) arm64 Apple M1
    Memory: 103.83 MB / 16.00 GB
    Shell: 5.1.8 - /opt/homebrew/Cellar/bash/5.1.8/bin/bash
  Binaries:
    Node: 18.14.2 - ~/.nvm/versions/node/v18.14.2/bin/node
    Yarn: 3.5.1 - ~/.npm_global/bin/yarn
    npm: 9.5.0 - ~/.nvm/versions/node/v18.14.2/bin/npm
    Watchman: 2023.03.27.00 - /opt/homebrew/bin/watchman
  Browsers:
    Brave Browser: 111.1.49.132
    Chrome: 113.0.5672.126
    Firefox: 109.0.1
    Safari: 16.1
  npmPackages:
    @babel/runtime: ^7.18.9 => 7.20.13
    @manypkg/cli: ^0.19.1 => 0.19.2
    @types/sharetribe-flex-sdk: https://github.com/journeyhorizon/sharetribe-flex-sdk-types#1.1.5 => 1.1.2
    @typescript-eslint/eslint-plugin: ^5.56.0 => 5.56.0
    eslint: ^8.36.0 => 8.36.0
    eslint-config-react: ^1.1.7 => 1.1.7
    eslint-config-react-native: ^4.1.0 => 4.1.0
    eslint-plugin-react-hooks: ^4.6.0 => 4.6.0
    prettier: ^2.7.1 => 2.8.3
    rome: 12.1.2 => 12.1.2
    turbo: ^1.4.2 => 1.7.0
    typescript: ^4.7.4 => 4.9.4
```

**Comment by natew** [comment]
Yo, you can re-open once sandbox is ready plz unless you can show me on the site or something. Or if it works in an older version lmk which one.

**Comment by matthewmturner** [comment]
@natew messaging you on discord with site info

**Comment by matthewmturner** [comment]
@natew ok I made sandbox https://codesandbox.io/p/sandbox/gallant-sky-0ge6cy?file=%2Fpackages%2Fapp%2Ffeatures%2Fhome%2Fform.tsx%3A72%2C43

site: https://0ge6cy-3000.csb.app/

I also noticed another issue where on small screen I can't open registration again after closing it by clicking overlay, you'll also see the issue on sandbox.

**Comment by matthewmturner** [comment]
@natew can we reopen now that I have sandbox?

**Comment by matthewmturner** [comment]
@natew has the team had a chance to look into this?

**Comment by matthewmturner** [comment]
Hey checking in here - could this be looked into soon?
